net/respondd/src/respondd.c

```diff
@@ -41,150 +48,381 @@
 #include <sys/socket.h>
 
 
+struct provider_list {
+	struct provider_list *next;
+
+	char *name;
+	respondd_provider provider;
+};
+
+struct request_type {
+	struct provider_list *providers;
+
+	struct json_object *cache;
+	uint64_t cache_time;
+	int64_t cache_timeout;
+};
+
+
+static int64_t now;
+
+
+static struct json_object * merge_json(struct json_object *a, struct json_object *b);
+
+
 static void usage() {
-  puts("Usage: respondd [-h] -g <group> -p <port> -i <if0> [-i <if1> ..]");
-  puts("  -g <ip6>         multicast group, e.g. ff02::2:1001");
-  puts("  -p <int>         port number to listen on");
-  puts("  -i <string>      interface on which the group is joined");
-  puts("  -c <string>      Lua command");
-  puts("  -h               this help\n");
+	puts("Usage: respondd [-h] -g <group> -p <port> -i <if0> [-i <if1> ..]");
```

Comment:
This line doesn't follow my understanding of help messages at all:
* `-h` is actually effectively mutually exclusive with all other options.
* This message suggests that `-g`, `-p` and `-i` are all mandatory, but none of them are.

I'd suggest the following:
```
Usage:
  respondd -h
  respondd [-p <port>] [-g <group> -i <if0> [-i <if1> ..]] [-d <dir>]
```